src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SelectVariants.java

```diff
@@ -0,0 +1,1244 @@
+            if (line instanceof VCFContigHeaderLine) {
+                continue; // skip old contig lines
+            }
+            if (line.getKey().equals(VCFHeader.REFERENCE_KEY)) {
+                continue; // skip the old reference key
+            }
+            lines.add(line);
+        }
+
+        lines.addAll(makeContigHeaderLines(refDict, referenceFile).stream().collect(Collectors.toList()));
+
+        final String referenceValue;
+        if (referenceFile != null) {
+            if (referenceNameOnly) {
+                final int extensionStart = referenceFile.getName().lastIndexOf(".");
+                referenceValue = extensionStart == -1 ? referenceFile.getName() : referenceFile.getName().substring(0, extensionStart);
+            }
+            else {
+                referenceValue = "file://" + referenceFile.getAbsolutePath();
+            }
+            lines.add(new VCFHeaderLine(VCFHeader.REFERENCE_KEY, referenceValue));
+        }
+        return lines;
+    }
+
+    /**
+     * Create VCFHeaderLines for each refDict entry, and optionally the assembly if referenceFile != null
+     * @param refDict reference dictionary
+     * @param referenceFile for assembly name.  May be null
+     * @return list of vcf contig header lines
+     */
+    private static List<VCFContigHeaderLine> makeContigHeaderLines(final SAMSequenceDictionary refDict,
+                                                                  final File referenceFile) {
+        final List<VCFContigHeaderLine> lines = new ArrayList<>();
+        final String assembly = referenceFile != null ? getReferenceAssembly(referenceFile.getName()) : null;
+        lines.addAll(refDict.getSequences().stream().map(contig -> makeContigHeaderLine(contig, assembly)).collect(Collectors.toList()));
+        return lines;
+    }
+
+    private static VCFContigHeaderLine makeContigHeaderLine(final SAMSequenceRecord contig, final String assembly) {
+        final Map<String, String> map = new LinkedHashMap<>(3);
+        map.put("ID", contig.getSequenceName());
+        map.put("length", String.valueOf(contig.getSequenceLength()));
+        if (assembly != null) {
+            map.put("assembly", assembly);
+        }
+        return new VCFContigHeaderLine(map, contig.getSequenceIndex());
+    }
+
+    /*
+     * TODO: the GATK code does this to include in the headers - seems pretty sketchy...
+     */
+    private static String getReferenceAssembly(final String refPath) {
+        // This doesn't need to be perfect as it's not a required VCF header line, but we might as well give it a shot
+        String assembly = null;
+        if (refPath.contains("b37") || refPath.contains("v37"))
+            assembly = "b37";
+        else if (refPath.contains("b36"))
+            assembly = "b36";
+        else if (refPath.contains("hg18"))
+            assembly = "hg18";
+        else if (refPath.contains("hg19"))
+            assembly = "hg19";
+        return assembly;
+    }
```

Comment:
Ack, that's horrible and should definitely die. But what's the functional implication of getting rid of this? Do we lose information in the header, or does this affect our ability to valid the reference dictionary? What was that info used for previously, do we know?